### DISCUSSION TITLE
AU Core AllergyIntolerance Consensus Discussion

### DISCUSSION THREAD

**Discussion by dt-r** [hl7au/au-fhir-core#40] [Consensus Discussions for R1]
Check out  [AU Core AllergyIntolerance](http://build.fhir.org/ig/hl7au/au-fhir-core/StructureDefinition-au-core-allergyintolerance.html), noting the draft [general conformance](http://build.fhir.org/ig/hl7au/au-fhir-core/general-requirements.html) and [Must Support](http://build.fhir.org/ig/hl7au/au-fhir-core/must-support.html) data obligations.

_To see the full constraints inherited from AU Base view the Key Elements Table_

AU Core AllergyIntolerance should support the minimum requirements to record, search, and fetch allergies/adverse reactions associated with a patient.

Comment on:
- If this profile was mandated tomorrow, what problems would that cause for your organisation?
- What's missing?
- Should anything change? What about the concepts you are using, is it sufficiently covered by the terminology in AU Core?

**Comment by jackieoc370** [comment]
Hi, wondering if this will enable information related to reactions to food with certain medications to be noted? For example, if taking X medication can not eat grapefruit or will cause allergic reaction. 
If not, can this please be slated as something essential to include for the future?

**Reply by gonatf** [reply]
There is a registered extension for this I think.  See https://build.fhir.org/ig/HL7/fhir-extensions/StructureDefinition-allergyintolerance-substanceExposureRisk.html

**Reply by brettesler-ext** [reply]
hi query on this one - is this a personal allergy for individuals or a general instruction for certain medications i.e. patient instructions on medication?

**Reply by dt-r** [reply]
Discussed in [2023-10-19 AU Core TDG Agenda/Minutes](https://confluence.hl7.org/pages/viewpage.action?pageId=184930419):
* Much discussion around the allergy vs intolerance vs interaction
* Need to be careful with setting hard rules in AU Core that would disagree with the usual practice i.e. if the doctor wants to include this in the patient record then it will appear in the Allergy list anyway
* Action to take that advice and come back to Jackie O’Connor.

**Comment by heathfrankel** [comment]
Would it be worth adding a recordedDate as MS. We have a recorder, so if a system has it, it may be helpful to also include the recordedDate. 
This provides an indication of currency, and could be used for clinical context related to the allergy but also in pre-population of Questionnaire scenarios.

**Reply by Pat-Redmond** [reply]
There is a field for 'onsetDateTime' which provides some context but probably not enough. 
Rather than 'recordedDate', I wonder if it would be worthwhile considering 'reviewedDate'. This would allow for the status to be reviewed and updated over time, while 'recordedDate' feels a bit more fixed.

**Reply by gonatf** [reply]
Would it be better to orient this around the asserter?  As in https://build.fhir.org/ig/HL7/fhir-extensions/StructureDefinition-allergyintolerance-assertedDate.html

**Reply by bhartisaroha** [reply]
Following up on this.
Should reveiwed Date and reviewer  be part of Au Core?

**Reply by dt-r** [reply]
I'm not convinced that recordedDate or reviewed date or reviewer should be supported in AU Core. Obviously we have use cases for this data but I don't think these elements are a) minimum requirements and b) across all use cases for allergy or intolerance information in the context of exchange. I note that these elements aren't supported by US Core, IPS or IPA. 

Happy to discuss and agree on how these should be done in FHIR separate to those elements forming conformance requirements in AU Core.

**Reply by dt-r** [reply]
recordedDate discussed in [2023-10-19 AU Core TDG Agenda/Minutes](https://confluence.hl7.org/pages/viewpage.action?pageId=184930419):
* Much discussion around the definition and use of recordedDate
* Need the relationship between recorded date and onset date
* Requires consideration on the meaning of MustSupport for systems that do not have this data
* Decision to park the AllergyIntolerance.recordedDate discussion until the discussion on generic must support definition in AU Core IG.

**Reply by dt-r** [reply]
Session on MustSupport held 15 December 2023 see zoom recording: https://hl7-au.zoom.us/rec/share/g94N1wl2SVwaoCSpLVGB8CsDBiuu1ZyCfPSB47uJy8HdNjqxA0DtERT441Ks7yw-.QxM4xcEHVruuO5zw
Zoom recording passcode: i^4vzrLR

Presentation Slides: [MustSupport-AUCoreTDG.pdf](https://confluence.hl7.org/download/attachments/204283377/MustSupport-AUCoreTDG.pdf?version=1&modificationDate=1702951016121&api=v2)

**Reply by dt-r** [reply]
I think we need to discuss what, if any, dates should be supported in AU Core for AU Core AllergyInterolance. I will take the action to ensure this put to the TDG. In the meantime, you can also provide a change proposal with rationale via raising a new [GitHub issue](https://github.com/hl7au/au-fhir-core/issues/new/choose).

**Comment by heathfrankel** [comment]
The search by clinical-status alone could be interpreted as a full population search, is this intended? 
Perhaps there needs to be be guidance around population health scenarios, or searches being performed within a patient/*.s scope
I suggest removing the mandatory requirement to support _revinclude of provenance until we at least have a profile for it.

**Reply by dt-r** [reply]
Search by clinical-status discussed in [2023-10-19 AU Core TDG Agenda/Minutes](https://confluence.hl7.org/pages/viewpage.action?pageId=184930419). Decision to park this for more discussion of search and what is in/out of scope for AU Core.

**Reply by dt-r** [reply]
Provenance to be progressed via https://github.com/hl7au/au-fhir-core/issues/70.

**Reply by dt-r** [reply]
@heathfrankel - guidance around pop health scenarios and the relevance to AU Core may not have been discussed. If you think this needs to be part of AU Core please raise a GitHub issue.

**Comment by dylan-jf7** [comment]
Need some clinical guidance on whether a recorded Allergy can outgrown and be no longer applicable. What would be the most appropriate way to record the status and the date of the change

**Reply by gonatf** [reply]
Something like this? https://build.fhir.org/ig/HL7/fhir-extensions/StructureDefinition-allergyintolerance-abatement.html

**Reply by dt-r** [reply]
Discussed in [2023-10-19 AU Core TDG Agenda/Minutes](https://confluence.hl7.org/pages/viewpage.action?pageId=184930419):
* AllergyIntolerance.clinicalStatus value ‘resolved’ can be used to represent an allergy that is not longer a concern ie outgrown
* Extension [Abatement](https://build.fhir.org/ig/HL7/fhir-extensions/StructureDefinition-allergyintolerance-abatement.html) would be appropriate to record the abatement date however most systems would not have this data
* No need to vote or consensus on this one, the group agreed no change to AU Core AllergyIntolerance is needed

**Comment by heathfrankel** [comment]
The AllergyIntolerance.code has a preferred binding inherited from AU BASE which is https://healthterminologies.gov.au/fhir/ValueSet/indicator-hypersensitivity-intolerance-to-substance-2.
This valueset has 177,120 terms derived from 

- < 418038007|Propensity to adverse reactions to substance| (0 terms - inactive, appears it should be 420134006? - 1409 terms)
- << 716186003|No known allergy| (7 terms)

and include valueset [https://healthterminologies.gov.au/fhir/ValueSet/adverse-reaction-agent-1](https://www.healthterminologies.gov.au/integration/R4/fhir/ValueSet?ui:filter=https%3A%2F%2Fhealthterminologies.gov.au%2Ffhir%2FValueSet%2Fadverse-reaction-agent-1) derived from
- ^ 32570211000036100 |Substance foundation reference set| (27577 terms)
- ^ 929360061000036106 |Medicinal product reference set| (2629 terms)
- ^ 929360081000036101 |Medicinal product pack reference set| (16143 terms)
- ^ 929360071000036103 |Medicinal product unit of use reference set| (7506 terms)
- ^ 929360021000036102 |Trade product reference set| (12834 terms)
- ^ 929360041000036105 |Trade product pack reference set| (42863 terms)
- ^ 929360031000036100 |Trade product unit of use reference set| (22160 terms)
- ^ 929360051000036108 |Containered trade product pack reference set| (45384 terms)
- ^ 142321000036106 |Adverse reaction agent reference set| (1090 terms)

My observation of a few primary care systems is that they have 4 ways of entering an allergy:

1. Non drug
2. Drug class
3. Drug ingredient
4. Drug product

Although number 4, Drug product, is quite common due to the product being in the prescription list and often a shortcut for being the cause of a reaction, a drug at the MPP, TPP or above would seem to be be too specific in most cases for an allergy. 
For a product, should we be standardising on a MP level code? This limits this set to only 2629 terms.

For number 3, we have a 'Adverse reaction agent reference set' (1090 terms) and 'Substance foundation reference set' (27577  terms), which both appear to have common ingredients amongst other substances (i.e. 1. Non drug). 
We also appear to have terms in these valueset that appear something like a drug class, but don't match the terms we have as drug classes in our primary care systems, Penicillins or Opioids and Narcotics.

It would be good to consider these commonly used groupings in systems as valuesets in their own right rather than these overly exhaustive mix of terms. In particular, I see the value in a standardised list of Drug Classes that can be used across all systems with appropriate coding.

If we look at the FHIR AllergyIntolerance category, there is another way of slicing these value sets that may be helpful, food, medication, environment and biological. It may be helpful to have recommended valuesets based on these categories and vendors may consider moving towards this classification of allergies rather than the 4 above, which do not align well? Currently, it would be difficult to populate the MS category element because it is either medication or one of the other 3 but not sure which one, it's just non drug.

Finally, the 'Propensity to adverse reactions to substance' is inactive. This should be fixed in the https://healthterminologies.gov.au/fhir/ValueSet/indicator-hypersensitivity-intolerance-to-substance-2 valueset.

420134006 Propensity to adverse reaction (1090 terms) appears to be the equivalent code (BTW, there are subitems of 'Propensity to adverse reaction to food' and 'Propensity to adverse reaction to drug'). 
However, these terms are duplicates of those already cover in the 'Adverse reaction agent reference', for example 'Allergy to penicillin (finding)' and 'Penicillin (substance)'. 
Do we need both? 
Which is more useful?
Which is correct in the context of AllergyIntolerance.code (is this a finding or a substance)?
What about AllergyIntolerance.reaction.substance?

In the end, why are we coding these allergies? In most cases it is for clinical decision support, and it would be easier to do this if we had one way to code our allergies, not two or many others.
I think we need to consider refining the current valueset to support our recommended best practice for doing this.
This could be extendible and grow this with implementation experience.

**Reply by dt-r** [reply]
Re: the 'Propensity to adverse reactions to substance' is inactive. This should be fixed in the https://healthterminologies.gov.au/fhir/ValueSet/indicator-hypersensitivity-intolerance-to-substance-2 valueset.

@liambarnes, @mjosborne1 and I have been working through that and will come back to the FHIR WG re AU Base with proposal for change probably in January.

**Reply by dt-r** [reply]
> Why are we coding these allergies?

Are we? Even if we are, the FHIR specifications don't force a code. Coding is supported but not required.

> In most cases it is for clinical decision support, and it would be easier to do this if we had one way to code our allergies, not two or many others.

Agreed. I hope that AUCDI will provide some advice on that for data entry and data use. In AU Core we can discuss if it makes sense to include that guidance in the exchange specification.

> This could be extendible and grow this with implementation experience.

The value set in AU Base is a starting point in FHIR supplied from the Agency (reflecting the Detailed Clinical Model work and clinical terminology analysis). I agree this can and should be better.

**Reply by dt-r** [reply]
@heathfrankel, maybe we should take this conversation to Zulip. It's not strictly raising a proposed change to the resource and more of a discussion that we want the community to be involved in and that's better had in Zulip.

**Reply by heathfrankel** [reply]
@dt-r https://healthterminologies.gov.au/fhir/ValueSet/indicator-hypersensitivity-intolerance-to-substance-2 has fixed this already or will be fixed?

**Reply by dt-r** [reply]
@heathfrankel, will be fixed

**Comment by bhartisaroha** [comment]
Looking for some recommendations on how to manage Reviewer and Reviewed Date against Allergy Intolerance resource.
Is an extension advisable?

**Reply by dt-r** [reply]
Hi @bhartisaroha, you should post this topic in Zulip: https://chat.fhir.org/#narrow/stream/179173-australia

**Reply by bhartisaroha** [reply]
Thank you @dt-r . I have posted in zulip also.

**Comment by dt-r** [comment]
This discussion is now closed. Feedback on AU Core should be raised using GitHub issues. Feedback might be suggesting a feature, pointing out a place where a specification is unclear or over-restrictive, identifying a spelling or grammar issue, identifying a defect, or suggesting that an entire area needs to be rethought.

Integration with HL7 JIRA is coming and when available specification feedback will be migrated to JIRA. An announcement will be posted in this discussion when it is available.

Questions on this profile can be raised in chat.fhir.org: https://chat.fhir.org/#narrow/stream/179173-australia
